frontend/src/metabase/dashboard/components/PublicLinkPopover/DashboardPublicLinkPopover.tsx

```diff
@@ -0,0 +1,49 @@
+import type { Dashboard } from "metabase-types/api";
+import { createPublicLink, deletePublicLink } from "metabase/dashboard/actions";
+import { useDispatch, useSelector } from "metabase/lib/redux";
+import { publicDashboard as getPublicDashboardUrl } from "metabase/lib/urls";
+import { getSetting } from "metabase/selectors/settings";
+import { PublicLinkPopover } from "./PublicLinkPopover";
+
+export const DashboardPublicLinkPopover = ({
+  dashboard,
+  target,
+  isOpen,
+  onClose,
+}: {
+  dashboard: Dashboard;
+  target: JSX.Element;
+  isOpen: boolean;
+  onClose: () => void;
+}) => {
+  const dispatch = useDispatch();
+
+  const uuid = dashboard.public_uuid;
+
+  const siteUrl = useSelector(state => getSetting(state, "site-url"));
+  const getPublicLink = () => {
+    if (!uuid) {
+      return null;
+    }
+    return getPublicDashboardUrl({ uuid, siteUrl });
+  };
```

Comment:
I kind of think instead of passing both `extensions` and `getPublicLink` in `QuestionPublicLinkPopover` would it make more sense to just pass something like `linkOptions` where each option has
- url
- name (used to display in the UI)

I feel like passing `getPublicLink` makes `PublicLinkPopover` very coupled with the component it's called from. `PublicLinkPopover` needs to know what arguments `getPublicLink` accepts to be able to pass them accordingly. But passing something like `linkOptions` could make it feel less coupled.

And if we only send 1 option we won't display the list of options to choose from (that's what this component does)

Reply:
I'll do this:
* Instead of using `getPublicLink`, I'll use the dashboard and question specific components to generate the link that will be displayed in the component
* For the `extensions`, I can still pass those into the `PublicLinkPopover`, but turn it into a controlled thing so the `PublicLinkPopover` doesn't deal with state.

This will allow `PublicLinkPopover` to be a completely display-driven component - its job will be to display the link and extensions if they exist, give the question/dashboard-specific components the selected options to create the link, and call the `create/delete` functions when its opened. It should have nothing to do with link formatting or construction